File: python/바이러스(2606).py
from collections import deque

a = int(input())
b = int(input())

graph = [[] for _ in range(a + 1)]

# 그래프 연결 정보 입력
for _ in range(b):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)  # 양방향 연결

def bfs(v):
    queue = deque([v])
    visited = [False] * (a+1)
    visited[v] = True
    cnt = 0

    while queue:
        node = queue.popleft()
        for n in graph[node]:
            if not visited[n]:
                visited[n] = True
                queue.append(n)
                cnt += 1
    print(cnt)

bfs(1)

File: js/바이러스(2606).js
const input = require('fs').readFileSync('dev/stdin').toString().trim().split('\n');

const N = +input.shift();
const M = +input.shift();

const graph = Array.from(Array(N+1), () => []);
const visited = Array.from(Array(N+1), () => false);
visited[1] = true;
let cnt = 0;

for (let i=0; i<M; i++) {
    const [first, second] = input[i].split(" ").map(Number);
    graph[first].push(second);
    graph[second].push(first);
}

function dfs(v) {
    for (let i of graph[v]) {
        if (visited[i] === false) {
            visited[i] = true;
            cnt++;
            dfs(i);
        }
    }
}

dfs(1);
console.log(cnt);